src/App/Fossa/LicenseScanner.hs

```diff
@@ -0,0 +1,172 @@
+  renderDiagnostic _ = "No native license scans were successful"
+
+runLicenseScanOnDir ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has Exec sig m
+  ) =>
+  Path Abs Dir ->
+  m [LicenseUnit]
+runLicenseScanOnDir scanDir = withThemisAndIndex $ themisRunner scanDir
+
+themisRunner ::
+  ( Has (Lift IO) sig m
+  , Has Diagnostics sig m
+  , Has Exec sig m
+  ) =>
+  Path Abs Dir ->
+  ThemisBins ->
+  m [LicenseUnit]
+themisRunner scanDir themisBins = runThemis themisBins scanDir
+
+runThemis :: (Has Exec sig m, Has Diagnostics sig m) => ThemisBins -> Path Abs Dir -> m [LicenseUnit]
+runThemis themisBins scanDir = do
+  context "Running license scan binary" $ execThemis themisBins scanDir
+
+md5 :: B.ByteString -> Digest MD5
+md5 = hash
+
+scanAndUploadVendoredDep ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has Logger sig m
+  , Has StickyLogger sig m
+  , Has Exec sig m
+  ) =>
+  ApiOpts ->
+  Path Abs Dir ->
+  VendoredDependency ->
+  m (Maybe Archive)
+scanAndUploadVendoredDep apiOpts baseDir vendoredDeps@VendoredDependency{..} = context "compressing and uploading vendored deps" $ do
+  logSticky $ "License Scanning '" <> vendoredName <> "' at '" <> vendoredPath <> "'"
+  vendoredDepDir <- case parseRelDir $ toString vendoredPath of
+    Left err -> fatalText ("Error constructing scan dir for vendored scan: " <> showT err)
+    Right val -> pure val
+  let cliScanDir = baseDir </> vendoredDepDir
+  themisScanResult <- runLicenseScanOnDir cliScanDir
+  case NE.nonEmpty themisScanResult of
+    Nothing -> Nothing <$ logError ("No license results found after scanning directory: " <> pretty (toString cliScanDir))
+    Just results -> uploadVendoredDep apiOpts vendoredDeps results
+
+uploadVendoredDep ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has StickyLogger sig m
+  ) =>
+  ApiOpts ->
+  VendoredDependency ->
+  NE.NonEmpty LicenseUnit ->
+  m (Maybe Archive)
+uploadVendoredDep apiOpts VendoredDependency{..} themisScanResult = do
+  let licenseSourceUnit =
+        LicenseSourceUnit
+          { licenseSourceUnitName = vendoredPath
+          , licenseSourceUnitType = CliLicenseScanned
+          , licenseSourceUnitLicenseUnits = themisScanResult
+          }
+
+  depVersion <- case vendoredVersion of
+    Nothing -> pure . showT . md5 . encodeUtf8 . show . NE.sort $ themisScanResult
+    Just version -> pure version
+
+  signedURL <- Fossa.getSignedLicenseScanURL apiOpts depVersion vendoredName
+
+  logSticky $ "Uploading '" <> vendoredName <> "' to secure S3 bucket"
+  void $ Fossa.licenseScanResultUpload signedURL licenseSourceUnit
+
+  pure $ Just $ Archive vendoredName depVersion
+
+-- | licenseScanSourceUnit receives a list of vendored dependencies, a root path, and API settings.
+-- Using this information, it license scans each vendored dependency, uploads the license scan results and then queues a build for the dependency.
+licenseScanSourceUnit ::
```

Comment:
[Optional] Let's move this to the top, so request flow follows the same chronological order as definition, so it's easy to read:

```
licenseScanSourceUnit
           | ->  scanAndUploadVendoredDep
                 |  ...
                 |      | ->  uploadVendoredDep 
                 |             | => getSignedLicenseScanURL (get s3 URL)
                 |             | => licenseScanResultUpload (upload to s3 url)
                 |
                 |      | => licenseScanFinalize (Queues build for licence scanned units?) 
```

I think an eventual doc diagram would help in the overall request flow.